### PR TITLE
Build macOS libraries in Travis CI

### PR DESCRIPTION
**Pull request by jkawamoto**
This PR provides Travis CI's configuration file to build macOS libraries in it. 

With this configuration file, when a tag is pushed to GitHub, Travis CI builds libstorj and dependent libraries, and then uploads them to GitHub releases as a tarballed file.

To upload files, the configuration file requires GitHub token (in line 55 of `.travis.yml`). To set the token, run 
```
travis setup releases
```

with the [Travis CI command line client](https://github.com/travis-ci/travis.rb#installation). 

Here is the [link to the detailed document](https://docs.travis-ci.com/user/deployment/releases/).

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -13,9 +13,8 @@ group = 'io.storj'
 version = '0.7.3'
 
 description = """Java bindings for libstorj"""
-
-sourceCompatibility = 1.7
-targetCompatibility = 1.7
+sourceCompatibility = 1.8
+targetCompatibility = 1.8
```
Comment (kaloyan-raev) [inline build.gradle]: @jkawamoto Is it really necessary to switch to Java 8?

Comment (kaloyan-raev) [reply build.gradle]: java-libstorj is used by android-libstorj. I am worried that this change my cause issues to some android devices. android-libstorj supports API 21+. It's seems that Java 8 is only supported by API 24+.

It should be possible JDK 1.8 to compile with 1.7 compatibility. How do you test this Travis configuration?

**Comment by jkawamoto** [comment]
I tested with JDK 1.7 and didn't get any errors. So, I updated the commit so that it can be used with 1.7.

**Comment by kaloyan-raev** [comment]
Great! Could you tell me how you test it locally, so I know how to do it in myself when I modify the `.travis.yml` in future?

**Comment by kaloyan-raev** [comment]
@jkawamoto This change broke the build on Linux, so I had to apply a fix: https://github.com/Storj/java-libstorj/commit/1b1dbdb8e6d73cdc88a4001097f914eb35a99d21

Please check if it still works on macOS.

**Comment by jkawamoto** [comment]
It looks working on macOS, too. https://travis-ci.org/jkawamoto/java-libstorj/builds/356835592?utm_source=github_status&utm_medium=notification

**Comment by kaloyan-raev** [comment]
I missed adding the Github OAuth token, so I added it with https://github.com/Storj/java-libstorj/commit/f7c7f7ebd1abdb811a48da0d6ea6af1607751c8a

Now we have a successful release build: https://travis-ci.org/Storj/java-libstorj/builds/357066973

The dylibs archive was automatically uploaded to the 0.8.1 release: https://github.com/Storj/java-libstorj/releases/tag/0.8.1

@jkawamoto Everything is as expected, right?

**Comment by jkawamoto** [comment]
Yes, it is. :)

**Pull request merged**